### PR TITLE
[x509] Fix comparison of certificate values

### PR DESCRIPTION
**Pull request by alxwr**
### What issues does this PR fix or reference?
Fixes an aspect of #56556

### Previous Behavior
Freshly created certificates are overwritten during the next run because the comparison of the certificate's values didn't work.

i.e.:

```                                                                    
              Status:                                                                                                 
                  ----------                                                                                          
                  New:                                                                                                
                      Certificate is valid and up to date                                                             
                  Old:                                                                                                
                      Certificate properties are different: Public Key
```

Acutally, the public key did not change!

### New Behavior

The certificate is not changed any more at very run of `salt`.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [X] Docs (none changed, this is a bugfix
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written

### Commits signed with GPG?
No

----

@glynnforrest @s0undt3ch This may interest you.

### CODE REVIEW

**Review by Ch3LL — CHANGES_REQUESTED** [review]
looks like there is a failing test that needs to be investigated

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/states/x509.py [hunk 1]
```diff
@@ -431,6 +429,9 @@ def _certificate_info_matches(cert_info, required_cert_info, check_serial=False)
 
     diff = []
     for k, v in required_cert_info.items():
+        # cert info comes as byte string
+        if type(v) == str:
```
Comment (s0undt3ch) [inline salt/states/x509.py:433]: Why type and not isinstance?

Comment (alxwr) [reply salt/states/x509.py:433]: Thanks, good catch!

File: CHANGELOG.md [hunk 2]
```diff
@@ -64,6 +64,7 @@ Fixed
 - Fixed macOS build scripts. (#57973)
 - Fixed Salt-API startup failure. (#57975)
 - Fixed CSR handling in x509 module (#54867)
+- Fixed comparison of certificate values
```
[the remark this answers is not in the document]
Comment (alxwr) [reply CHANGELOG.md:67]: Thanks, didn't know that! (TIL) :-)

**Comment by alxwr** [comment]
@Ch3LL @s0undt3ch I added your requested changes. Please review! Thanks! :-)

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
looks like you have a merge conflict that needs to be cleaned up

**Comment by alxwr** [comment]
@Ch3LL Please review again! :-)

**Review by Ch3LL — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by alxwr** [comment]
@dwoz  Thanks! 🎊

**Pull request merged**